source/MaterialXGenShader/ShaderGraph.cpp

```diff
@@ -55,7 +55,7 @@ void ShaderGraph::addInputSockets(const InterfaceElement& elem, GenContext& cont
                 inputSocket = addInputSocket(port->getName(), portType);
                 if (!portValue.empty())
                 {
-                    inputSocket->setValue(port->getValue());
+                    inputSocket->setValue(port->getResolvedValue());
```

Comment:
We computed the resolved token for enumerations (line 44) but didn't use resolved values when no enums.

Reply:
Would it be possible to resolve only once? For example above calling 
```
ValuePtr value = port->getResolvedValue();
const string portValue = value->getValueString();
```
to get the value both as a ValuePtr and a string, but only resolve once.

Reply:
Good idea. I'll fix.